### DISCUSSION TITLE
About rewriting some service default implementations

### DISCUSSION THREAD

**Discussion by GreenShadeZhang** [elsa-workflows/elsa-core#1294] [Q&A]
Can these services provide rewriting capability? Or do you have this plan? Do you have any needs in this respect.

![image](https://user-images.githubusercontent.com/24240675/126460865-2a2fd1fd-25d8-4d03-9e87-9b16105dc387.png)

**Comment by sfmskywalker** [comment]
Yes, you can provide your own implementations for any of these services. They all implement an interface that you can implement yourself.
If you want to e.g. provide your own `GetStartActivitiesProvider` implementation, do these things:

1. Create your own class that implements `IGetsStartActivities`
2. Replace the existing service registration:`services.Replace<IGetsStartActivities, MyCustomGetsStartActivities>(ServiceLifetime.Transient);`
